packages/aila/src/core/chat/AilaStreamHandler.ts

```diff
@@ -190,6 +198,46 @@ export class AilaStreamHandler {
           return m;
         }) as { role: "user" | "assistant"; content: string }[],
       onUpdate: streamHandler, // This is the new part
+      customAgents: {
+        mathsStarterQuiz: async ({ document }) => {
+          const quiz = await this._chat.fullQuizService.createBestQuiz(
+            "/starterQuiz",
+            document,
+          );
+
+          return quiz;
+        },
+        mathsExitQuiz: async ({ document }) => {
+          const quiz = await this._chat.fullQuizService.createBestQuiz(
+            "/exitQuiz",
+            document,
+          );
+
+          return quiz;
+        },
+        fetchRagData: async ({ document }) => {
+          if (this._chat.relevantLessons) {
+            const results = await getRagLessonPlansByIds({
```

Comment:
If we already have relevantLessons (ids) on the plan , then fetch the lesson plans.

Reply:
Otherwise get the relevant lesson plans from RAG